Match features

DvirDukhan opened the pull request:
this PR re-categorize `MATCH` clause feature files 

* `Match1`: Match nodes
** This group is about matching nodes.
* `Match2`: Match relationships
** This group is about matching relationships.
* `Match3`: Match patterns
** This group is about matching patterns.
* `Match4`: Match variable length patterns
** This group is about matching variable length patterns.
* `Match5`: Match variable length patterns over given graphs
** This group is about matching patterns over given graphs.
* `Match6`: Match named paths
** This group is about matching named paths.
* `Match7`: Optional match
** This group is about optional matching with all of the above.
* `Match8`: Miscellaneous features
** This group is about miscellaneous `MATCH` clause features.
* `Match9`: Deprecated features
** This group is about deprecated `MATCH` clause features.
* `Match10`: Negative tests
** This group is about failure for the `MATCH` clause.

DvirDukhan commented on tck/features/match/Match1.feature:
```diff
@@ -0,0 +1,277 @@
+      | 1 |
+    And no side effects
+
+  Scenario: Simple node property predicate
```
duplicated. Created a scenario with an inline property filter. The explicit pre moved to WHERE feature.

DvirDukhan commented on tck/features/match/Match2.feature:
```diff
@@ -0,0 +1,213 @@
+
+Feature: Match2
+
+  Scenario: Do not return non-existent relationships
```
The first scenario in Match1.feature tests the same for nodes.
renamed to "Match non-existent relationships returns empty"

DvirDukhan commented on tck/features/match/Match2.feature:
```diff
@@ -0,0 +1,213 @@
+      | [:T1] |
+    And no side effects
+
+  Scenario: Matching a self-loop
```
added

DvirDukhan commented on tck/features/match/Match3.feature:
```diff
@@ -0,0 +1,915 @@
+      | (:D {animal: 'cow'})    | (:B {animal: 'cow'})    |
+    And no side effects
+
+  Scenario: Rel type function works as expected
```
renamed to "Match with relationship type predicate"

DvirDukhan commented on tck/features/match/Match3.feature:
```diff
@@ -0,0 +1,915 @@
+      | ({name: 'Andres'}) |
+    And no side effects
+
+  Scenario: Matching using a simple pattern with label predicate
```
moved to WHERE

DvirDukhan commented on tck/features/match/Match4.feature:
```diff
@@ -0,0 +1,274 @@
+      | (:A)  | (:C)   |
+    And no side effects
+
+  Scenario: Matching relationships into a list and matching variable length using the list, with bound nodes
```
moved to deprecated

DvirDukhan commented on tck/features/match/Match4.feature:
```diff
@@ -0,0 +1,274 @@
+      | (:A)  | (:C)   |
+    And no side effects
+
+  Scenario: Matching relationships into a list and matching variable length using the list, with bound nodes, wrong direction
```
moved to deprecated

DvirDukhan commented on tck/features/match/Match7.feature:
```diff
@@ -0,0 +1,385 @@
+      | 'Mark'  |
+    And no side effects
+
+  Scenario: Matching from null nodes should return no results owing to finding no matches
```
Move to match3

DvirDukhan commented on tck/features/match/Match7.feature:
```diff
@@ -0,0 +1,385 @@
+      | b |
+    And no side effects
+
+  Scenario: Matching from null nodes should return no results owing to matches being filtered out
```
moved to match3

DvirDukhan commented on tck/index.adoc:
```diff
@@ -26,7 +26,26 @@ Each feature follows a naming convention of `<clause-name><group-id>.feature`.
+** This group is about matching named paths.
+* `Match7`: Optional match
+** This group is about optional matching with all of the above.
+* `Match8`: Miscellaneous features
```
changed

DvirDukhan commented on tck/features/match/Match8.feature:
```diff
@@ -0,0 +1,263 @@
+      | 'Someone' |
+    And no side effects
+
+  Scenario: Cope with shadowed variables
```
renamed to "Return bounded node property"

DvirDukhan commented on tck/features/match/Match8.feature:
```diff
@@ -0,0 +1,263 @@
+      | 'King Kong'  |
+    And no side effects
+
+  Scenario: Do not fail when evaluating predicates with illegal operations if the AND'ed predicate evaluates to false
```
moved to pattern matching - where

DvirDukhan commented on tck/features/match/Match8.feature:
```diff
@@ -0,0 +1,263 @@
+      | (:TextNode {var: 'text'}) |
+    And no side effects
+
+  Scenario: Do not fail when evaluating predicates with illegal operations if the OR'd predicate evaluates to true
```
moved to pattern matching - where

DvirDukhan commented on tck/features/match/Match1.feature:
```diff
@@ -0,0 +1,277 @@
+
+Feature: Match1
+
+  Scenario: Do not return non-existent nodes
```
I will add the scenario numbering once the feature files structure is approved. otherwise, it will be continuous update of the numbering.

DvirDukhan commented:
@hvub 
PR comments addressed.
For each feature file with "major group number" I added feature files with "minor group number" so that it will handle specific clauses. For example, major number 1 is for nodes matching scenarios, so feature file `Match1` will contain basic node matching scenarios. The feature file `Match1-1` is for match nodes `RETURN` clause scenarios.
The keys for the minor number are:
1 - `RETURN` clause scenarios
2 - `WHERE` clause scenarios
3 - `WITH` clause scenarios
4 - built-in functions

hvub commented:
Hi @DvirDukhan,

thanks for the changes.

One comment, I may have forgotten to mention in the review:

The intended folder structure is the following:

```
- features
   - clauses
      - create
      - match
      - where
      - return
      - ...
   - expressions
      - arithmetics
      - builtin functions
      - literals
      - ...
```

To meet this folder structure, can you please move your `match` folder under the `clauses`. Thanks.

hvub commented:
A second point: When I said in my comments that a scenario looks like a WHERE scenario, I meant it should end up in `features/clauses/where`. 

However I think, your minor groups are a helpful intermediate step to that, because it already memorizes that we have identified a scenario as being potentially in a certain category, although we are not working on that category yet. In other words, it will simplify later work, such as categorizing WHERE. 

I wonder, though if we should move these `MatchX-Y.feature` files into a kind of staging folder, say `features/precategorized`, to keep the work that went in to creating them, but also make clear that the scenarios they contain are intended end up somewhere else, e.g. in `features/clauses/where/WhereX.feature` files ones we get to categorizing WHERE. Anyway, we can discuss that on Thursday in our regular meeting.

hvub commented on tck/features/match/Match2-2.feature:
```diff
@@ -0,0 +1,71 @@
+
+#encoding: utf-8
+
+Feature: Match2-1 - Match relationships WHERE clause scenarios
```
filename is Match2-2

hvub commented on tck/features/match/Match3-3.feature:
```diff
@@ -0,0 +1,86 @@
+
+#encoding: utf-8
+
+Feature: Match3-2 - Match fixed length patterns WITH clause scenarios
```
filename is Match3-3

hvub commented on tck/features/match/Match4-2.feature:
```diff
@@ -0,0 +1,52 @@
+
+#encoding: utf-8
+
+Feature: Match4-4 - Match variable length patterns WHERE clause scenarios
```
filename is Match4-2

hvub commented on tck/features/match/Match7-2.feature:
```diff
@@ -0,0 +1,70 @@
+
+#encoding: utf-8
+
+Feature: Match7 - Optional match RETURN clause scenarios
```
filename is Match7-2

hvub commented on tck/features/match/Match3.feature:
```diff
@@ -0,0 +1,520 @@
+      | b             |
+      | (:C {num: 3}) |
+    And no side effects
+
```
Please remove extra blank line

DvirDukhan commented:
@hvub 
PR comments fixed

regarding the new and deleted scenarios:
1. Some of the deleted scenarios are explicit or implicit duplications of other scenarios, so I removed them.
2. New scenarios are renamed scenarios or scenarios which are related to different clause type, as is, but with a little modification they can be part of match clause tests, for example: converting explicit `WHERE` clause to an inline filter.
changes:
```
MatchAcceptance.feature:Filter based on rel prop name => Match2.featue:match relationship with inline property value 
MatchAcceptance2.feature:Matching a self-loop + MatchingSelfRelationships.feature:Undirected match of self-relationship in self-relationship graph=> Match2.feature:Matching a self-loop with an undirected relationship pattern
MatchingSelfRelationships.feature:Directed match on self-relationship graph => Match2.feature:Matching a self-loop with a directed relationship pattern
MatchAcceptance.feature:Walk alternative relationships => copy and modified to Match2.feature:Match relationships with multiple types
MatchAcceptance.feature:Simple node property predicate => copy and modified to Match1.feature:Simple node inline property predicate
OptionalMatchAcceptance:feature:OPTIONAL MATCH and `collect() => Match7-4.feature:OPTIONAL MATCH and collect() on node property
```

hvub commented:
This comment is mainly concerned with a more detailed investigation on scenarios found by the Diff tool to be missing.

**Removed scenarios with acceptable reason:**

* Removed: [`uncategorized :: "MatchAcceptance2" :: "OPTIONAL MATCH returns null"`](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L954)
  * Reason: identical with [`clauses/match :: "Match7 - Optional match scenarios" :: "Simple OPTIONAL MATCH on empty graph"`](https://github.com/opencypher/openCypher/pull/387/files#diff-c798c3cedb1f755db1567273da1a91bfR33)

**Removed scenarios that need adjustment:**

* Removed: [`uncategorized :: "OptionalMatchAcceptance" :: "OPTIONAL MATCH and `collect()`"`](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L322)
   * Reason: Not actually removed, but simplified to [``precategorized/expressions/build_in_functions :: "Feature: Match7-4 - Optional Match build in functions scenarios" :: "OPTIONAL MATCH and `collect() on node property\`"``](https://github.com/opencypher/openCypher/pull/387/files#diff-117086a4bd9bf64be39b09f088af6da1R50)
   * Note that [Background](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L33) was intentionally or unintentionally dropped from the scenario when moved into new file. However, the background appears to be irrelevant for the essence of the scenario. Hence, it is an acceptable change.
   * **Recommended action:**
     * The scenario title should change to ``OPTIONAL MATCH and `collect()` on node property`` (note changed position of the second backtick)

**Scenarios removed without acceptable reason:**

*  [`uncategorized :: "OptionalMatch" :: "Satisfies the open world assumption, relationships between different nodes"`](https://github.com/opencypher/openCypher/pull/387/files#diff-94d23717bfbc188c2c10b0a26bb13584L70)
*  [`uncategorized :: "OptionalMatch" :: "Satisfies the open world assumption, relationships between same nodes"`](https://github.com/opencypher/openCypher/pull/387/files#diff-94d23717bfbc188c2c10b0a26bb13584L33)
*  [`uncategorized :: "OptionalMatch" :: "Satisfies the open world assumption, single relationship"`](https://github.com/opencypher/openCypher/pull/387/files#diff-94d23717bfbc188c2c10b0a26bb13584L52)

**Recommended action:** I preferred to have these scenarios being put back into `Match7 - Optional match scenarios`. They have some overlap with others there and that could be tidied up at some put. Nevertheless, at these point I like to focus on categorizes with otherwise minimal changes. This way, the categorization task itself as well as reviewing will get simpler.

**Misc**

* [`tck/features/precategorized/expressions/build_in_functions/Match7-4.feature`](https://github.com/opencypher/openCypher/pull/387/files#diff-117086a4bd9bf64be39b09f088af6da1R67) appear to miss a new line at its end. 

**To be done**
* Detailed investigation on scenarios found by the Diff tool to be added
* Detailed investigation on scenarios found by the Diff tool to be modified
* Correlate investigation with scenarios @DvirDukhan [mentioned explicitly](https://github.com/opencypher/openCypher/pull/387#issuecomment-582500341) as changed

hvub commented on tck/features/precategorized/clauses/where/Match1-2.feature:
```diff
@@ -0,0 +1,86 @@
+
+#encoding: utf-8
+
+Feature: Match 1-2 - Match nodes WHERE clause scenarios
```
This has an extra space, should be `MATCH1-2` instead.

hvub commented on tck/features/precategorized/clauses/with/Match1-3.feature:
```diff
@@ -0,0 +1,50 @@
+
+#encoding: utf-8
+
+Feature: Match 1-3 - Match nodes WITH clause scenarios
```
This has an extra space, should be `Match1-3` instead.

hvub commented on tck/features/precategorized/clauses/return/Match1-1.feature:
```diff
@@ -0,0 +1,220 @@
+
+#encoding: utf-8
+
+Feature: Match 1-1 - Match nodes RETURN clause scenarios
```
This has an extra space, should be `Match1-1` instead.

hvub commented on tck/features/precategorized/expressions/build_in_functions/Match6-4.feature:
```diff
@@ -0,0 +1,81 @@
+
+#encoding: utf-8
+
+Feature: Feature: Match6-4 - Match named paths built in functions scenarios
```
Contains an extra `Feature: ` in its name. Please remove.

hvub commented:
This comment is mainly concerned with a more detailed investigation on scenarios found by the Diff tool to be added.

* [clauses/match :: "Match1 - Match Nodes scenarios" :: "Simple node inline property predicate"](https://github.com/opencypher/openCypher/pull/387/files#diff-86ca245b59cb0ed247dfa8660e06e51aR78)
   * Reason for adding: Existed only in a [`WHERE` variant](https://github.com/opencypher/openCypher/pull/387/files#diff-aa01aa609920341c08f95433da032664R33) before.
   * Proposed change: Add two extra nodes to the graph, i.e. change the second step (`And having executed`) to include: `CREATE ({name: 'monkey'})` and `CREATE ({firstname: 'bar'})`.
* [clauses/match :: "Match2 - Match relationships scenarios" :: "Match relationship with inline property value"](https://github.com/opencypher/openCypher/pull/387/files#diff-3ffb8380026dff2ad5db6db2b920d29dR97)
   * Reason for adding: Existed only in a [`WHERE` variant](https://github.com/opencypher/openCypher/pull/387/files#diff-2ce56eea9d0826210d360d930cf7a45aR33) before.
* [clauses/match :: "Match2 - Match relationships scenarios" :: "Match relationship with inline property value"](https://github.com/opencypher/openCypher/pull/387/files#diff-3ffb8380026dff2ad5db6db2b920d29dR113)
   * Reason for adding: Existed only in a [`WHERE` variant](https://github.com/opencypher/openCypher/pull/387/files#diff-2ce56eea9d0826210d360d930cf7a45aR50) before.
* [clauses/match :: "Match2 - Match relationships scenarios" :: "Matching a self-loop with a directed relationship pattern"](https://github.com/opencypher/openCypher/pull/387/files#diff-3ffb8380026dff2ad5db6db2b920d29dR80)
   * Reason for adding: Seems to be add for completeness in the feature.
* [``precategorized/expressions/build_in_functions :: "Feature: Match7-4 - Optional Match build in functions scenarios" :: "OPTIONAL MATCH and `collect() on node property`"``](https://github.com/opencypher/openCypher/pull/387/files#diff-117086a4bd9bf64be39b09f088af6da1R50)
   * Reason for adding: Not actually added, but simplified version of [`uncategorized :: "OptionalMatchAcceptance" :: "OPTIONAL MATCH and `collect()`"`](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L322)

** Misc **
[`Feature: Feature: Match6-2 - Match named paths WHERE clause scenarios`](https://github.com/opencypher/openCypher/pull/387/files#diff-160cf1185840a9e786b661fe031e5b5dR31) contains an extra `Feature: ` in its name. Please remove.

**To be done**
* Detailed investigation on scenarios found by the Diff tool to be modified
* Correlate investigation with scenarios @DvirDukhan [mentioned explicitly](https://github.com/opencypher/openCypher/pull/387#issuecomment-582500341) as changed

hvub commented:
This comment is mainly concerned with a more detailed investigation on scenarios found by the Diff tool to be changed.

**Changed scenarios**

* Before: [uncategorized⟩⟩MatchAcceptance :: Cope with shadowed variables](https://github.com/opencypher/openCypher/pull/387/files#diff-2729dab85c6d1cf47c9e6046bdca99dbL161)
  After: [precategorized⟩clauses⟩with⟩⟩Match1-3 - Match nodes WITH clause scenarios :: Return bounded node property](https://github.com/opencypher/openCypher/pull/387/files#diff-51261b881d94c37b31f2a1d5d024dc27R33)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance :: Rel type function works as expected](https://github.com/opencypher/openCypher/pull/387/files#diff-2729dab85c6d1cf47c9e6046bdca99dbL292)
  After: [precategorized⟩clauses⟩where⟩⟩Match3-2 - Match fixed length patterns WHERE clause scenarios :: Match with relationship type predicate](https://github.com/opencypher/openCypher/pull/387/files#diff-4bb23063c5615841b2d0d074f132d18cR128)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance :: Walk alternative relationships](https://github.com/opencypher/openCypher/pull/387/files#diff-2729dab85c6d1cf47c9e6046bdca99dbL313)
  After: [precategorized⟩clauses⟩where⟩⟩Match2-2 - Match relationships WHERE clause scenarios :: Walk alternative relationships](https://github.com/opencypher/openCypher/pull/387/files#diff-2ce56eea9d0826210d360d930cf7a45aR50)
  * Detected: Moved, StepsChanged
  * Comment: Removes some whitespace in the Cypher query of the 2nd step (`And having executed`). 
  * **Suggestion:** Put whitespace back in, so that the scenario does not show up as changed.
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Counting an empty graph](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L1323)
  After: [precategorized⟩clauses⟩return⟩⟩Match1-1 - Match nodes RETURN clause scenarios :: Return count aggregation over an empty graph](https://github.com/opencypher/openCypher/pull/387/files#diff-cbc61a0718f2552f18067d7b220d2f67R130)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Do not return non-existent nodes](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L33)
  After: [clauses⟩match⟩⟩Match1 - Match Nodes scenarios :: Match non-existent nodes returns empty](https://github.com/opencypher/openCypher/pull/387/files#diff-86ca245b59cb0ed247dfa8660e06e51aR33)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Do not return non-existent relationships](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L44)
  After: [clauses⟩match⟩⟩Match2 - Match relationships scenarios :: Match non-existent relationships returns empty](https://github.com/opencypher/openCypher/pull/387/files#diff-3ffb8380026dff2ad5db6db2b920d29dR33)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Matching a self-loop](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L1843)
  After: [clauses⟩match⟩⟩Match2 - Match relationships scenarios :: Matching a self-loop with an undirected relationship pattern](https://github.com/opencypher/openCypher/pull/387/files#diff-3ffb8380026dff2ad5db6db2b920d29dR63)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Matching with aggregation](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L998)
  After: [precategorized⟩clauses⟩return⟩⟩Match1-1 - Match nodes RETURN clause scenarios :: Return count aggregation over nodes](https://github.com/opencypher/openCypher/pull/387/files#diff-cbc61a0718f2552f18067d7b220d2f67R142)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩MatchAcceptance2 :: Optionally matching named paths](https://github.com/opencypher/openCypher/pull/387/files#diff-d258d3042007565bc4236dc2c50c53b8L326)
  After: [clauses⟩match⟩⟩Match7 - Optional match scenarios :: Optionally matching named paths - existing result](https://github.com/opencypher/openCypher/pull/387/files#diff-c798c3cedb1f755db1567273da1a91bfR289)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩OptionalMatchAcceptance :: Named paths in optional matches](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L122)
  After: [clauses⟩match⟩⟩Match7 - Optional match scenarios :: Optionally matching named paths - null result](https://github.com/opencypher/openCypher/pull/387/files#diff-c798c3cedb1f755db1567273da1a91bfR267)
  * Detected: PotentiallyRenamed
  * Comment: Indeed just renamed. **Okay.**
* Before: [uncategorized⟩⟩OptionalMatchAcceptance :: OPTIONAL MATCH and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L340)
  After: [precategorized⟩clauses⟩where⟩⟩Match7-2 - Optional match WHERE clause scenarios :: OPTIONAL MATCH and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-f92d112ad67e15ba37d184f3bde9f0a2R161)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L33), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. However, the missing input data appears irrelevant for the result of the scenario's test query. The `MATCH` patterns only query node labels (`X`, `Y`) and a relationship type (`E1`), which are not present in the missing data. The `WHERE` clause filter on a property `val`, which is not present in the missing data. **Okay.**
* Before: [uncategorized⟩⟩OptionalMatchAcceptance :: OPTIONAL MATCH on two relationships and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L362)
  After: [precategorized⟩clauses⟩where⟩⟩Match7-2 - Optional match WHERE clause scenarios :: OPTIONAL MATCH on two relationships and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-f92d112ad67e15ba37d184f3bde9f0a2R184)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L33), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. However, the missing input data appears irrelevant for the result of the scenario's test query. The `MATCH` patterns only query node labels (`X`, `Y`, `Z`) and relationship types (`E1`, `E2`), which are not present in the missing data. The `WHERE` clause filter on a property `val`, which is not present in the missing data. **Okay.**
* Before: [uncategorized⟩⟩OptionalMatchAcceptance :: OPTIONAL MATCH with labels on the optional end node](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L146)
  After: [clauses⟩match⟩⟩Match7 - Optional match scenarios :: OPTIONAL MATCH with labels on the optional end node](https://github.com/opencypher/openCypher/pull/387/files#diff-c798c3cedb1f755db1567273da1a91bfR419)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L33), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. However, the missing input data appears irrelevant for the result of the scenario's test query. The `MATCH` pattern only queries node labels (`X`, `Y`), which are not present in the missing data. **Okay.**
* Before: [uncategorized⟩⟩OptionalMatchAcceptance :: Two OPTIONAL MATCH clauses and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L384)
  After: [precategorized⟩clauses⟩where⟩⟩Match7-2 - Optional match WHERE clause scenarios :: Two OPTIONAL MATCH clauses and WHERE](https://github.com/opencypher/openCypher/pull/387/files#diff-f92d112ad67e15ba37d184f3bde9f0a2R207)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/pull/387/files#diff-019338a9f6d0444c704968e2a6e5b9c5L33), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. However, the missing input data appears irrelevant for the result of the scenario's test query. The `MATCH` patterns only query node labels (`X`, `Y`, `Z`) and relationship types (`E1`, `E2`), which are not present in the missing data. The `WHERE` clause filter on a property `val`, which is not present in the missing data. **Okay.**
* Before: [uncategorized⟩⟩VarLengthAcceptance :: Fail on negative bound](https://github.com/opencypher/openCypher/pull/387/files#diff-156921a38ebae60e6f235d2637ffe8b5L243)
  After: [clauses⟩match⟩⟩Match10 - Match clause failure scenarios :: Fail on negative bound](https://github.com/opencypher/openCypher/pull/387/files#diff-883936f612e18d8fbcb79ebcf7749a82R43)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L34), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. The scenario tests for failing on a syntax error. It does so originally under the presence of relevant data. It seems better to keep it like that. 
  * **Suggestion:** Put the data from the original `Background:` declaration into the scenario. Precisely, insert original lines [36](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L36)–[67](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L67) in the scenario after line [44](https://github.com/opencypher/openCypher/pull/387/files#diff-883936f612e18d8fbcb79ebcf7749a82R44).
* Before: [uncategorized⟩⟩VarLengthAcceptance :: Fail when asterisk operator is missing](https://github.com/opencypher/openCypher/pull/387/files#diff-156921a38ebae60e6f235d2637ffe8b5L119)
  After: [clauses⟩match⟩⟩Match10 - Match clause failure scenarios :: Fail when asterisk operator is missing](https://github.com/opencypher/openCypher/pull/387/files#diff-883936f612e18d8fbcb79ebcf7749a82R33)
  * Detected: Moved, StepsChanged
  * Comment: Original feature file contained a [`Background:` declaration](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L34), which is not present in the new feature file. Consequentially, the input data for the scenario has changed. The scenario tests for failing on a syntax error. It does so originally under the presence of relevant data. It seems better to keep it like that. 
  * **Suggestion:** Put the data from the original `Background:` declaration into the scenario. Precisely, insert original lines [36](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L36)–[67](https://github.com/opencypher/openCypher/blob/master/tck/features/uncategorized/VarLengthAcceptance.feature#L67) in the scenario after line [34](https://github.com/opencypher/openCypher/pull/387/files#diff-883936f612e18d8fbcb79ebcf7749a82R34).

**To be done**

* Correlate investigation with scenarios @DvirDukhan [mentioned explicitly](https://github.com/opencypher/openCypher/pull/387#issuecomment-582500341) as changed

hvub commented:
From my point of view this PR can be merged. Let's decide in our next meeting.
